### PR TITLE
Possibility to reset connection to redis

### PR DESCRIPTION
**Pull request by vdoleans01**

### CODE REVIEW

File: session-replacement/src/main/java/com/amadeus/session/repository/redis/RedisSessionRepository.java [hunk 1]
```diff
@@ -530,4 +541,18 @@ public void sessionIdChange(SessionData sessionData) {
...
     expirationManager.sessionIdChange(sessionData);
   }
+
+  @Override
+  public boolean isConnected() {
+    try {
+      redis.info("server");
+      return true;
+    }
+    catch ( Exception e ) {
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/repository/redis/RedisSessionRepository.java]: Are there no other ways to detect that Redis is disconnected than by catching an exception?

I would rather we find another way. If not possible, then can we at least be more specific on the Exception caught?

File: session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * The class is used to track the number of item during a time.
+ * input parameter
+ *  delay , time in milliseconds where items are keep;
+ *  max: if the number if items is more than the max value the ErrorTracker reachLimits   
+ *
+ */
+
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java]: Better to remove the useless end of line :)

File: session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package com.amadeus.session;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * The class is used to track the number of item during a time.
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java]: during a time?

File: session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ * input parameter
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java]: *  @param delay 
                  Time in milliseconds while items are kept
*  @param max
                   If the number of items is more than the max value the Tracker is considered in error

File: session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  public ErrorTracker(int delay , int max) {
+    super();
+    this.delay = delay;
+    this.max = max;
+  }
+
+  private ConcurrentLinkedQueue<Long> list = new ConcurrentLinkedQueue<Long>();
+
+  /**
+   * The class calculate the items during a  time
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java]: calculates

Could you clarify "during a time"?

File: session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+   */
+  final int delay ;
+  /**
+   * After the limits the Tracker is considering in error
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java]: considered

File: session-replacement/src/main/java/com/amadeus/session/ResetManager.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  
+  private final SessionConfiguration configuration;
+  protected final ExecutorFacade executors;
+
+  public ResetManager(ExecutorFacade executors , SessionConfiguration configuration ) {
+    this.executors = executors;
+
+    monitoring = new MetricRegistry ();
+    
+    String SESSIONS_METRIC_MANAGER_PREFIX = "com.amadeus.session.manager";
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ResetManager.java]: You could put is as a constant

File: session-replacement/src/main/java/com/amadeus/session/repository/redis/NotificationExpirationManagement.java [hunk 8]
```diff
@@ -524,6 +524,36 @@ public void close() {
     }
   }
 
+  public void reset() {
+		if (expirationListener != null) {
+
+			try {
+				expirationListener.close(redis);
+				expirationListener = null;
+			} catch (Exception e) {
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/repository/redis/NotificationExpirationManagement.java]: Cannot we be more specific on the Exception caught?
Same remark for the other cases where we caught Exception

Small spelling error: generates errors

File: session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+   */
+  final int max ;
+  
+  public void addError( long now )  {
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java]: Maybe you could add some javadoc explaining the method?

File: session-replacement/src/main/java/com/amadeus/session/ResetManager.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    String ResetManager_created_METRIC = name(SESSIONS_METRIC_MANAGER_PREFIX, "initialized");
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ResetManager.java]: Would put it, and the others as constants (and hence all in upper case too)

File: session-replacement/src/main/java/com/amadeus/session/ResetManager.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+
+public class ResetManager {
+  private final MetricRegistry monitoring;
+  private JmxReporter reporter; 
+  private final ErrorTracker errorTracker ;
+  
+   
+  private final Meter create_metrics ;
```
Comment (mgarcez) [inline session-replacement/src/main/java/com/amadeus/session/ResetManager.java]: Would either put the variable name as camelCase, or as a CONSTANT_NAME.
I would also replace "metrics" by "meter", or "metric" in singular.

File: session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java (hunk 9 above)
Comment (vdoleans01) [reply session-replacement/src/main/java/com/amadeus/session/ErrorTracker.java]: fixed

File: session-replacement/src/main/java/com/amadeus/session/ResetManager.java (hunk 10 above)
Comment (vdoleans01) [reply session-replacement/src/main/java/com/amadeus/session/ResetManager.java]: fixed

File: session-replacement/src/main/java/com/amadeus/session/ResetManager.java (hunk 11 above)
Comment (vdoleans01) [reply session-replacement/src/main/java/com/amadeus/session/ResetManager.java]: fixed

**Pull request merged**